Re-enable integration tests for cpp

JochenKienzle opened the pull request:
- solve relative path issues (pg-config, adapters)
- copy pg-config files for basic example
- adapt paths and links (testset vs. testsuite)
- adapt pg-config for testsuites
- add build folders to dockerignore to avoid pre-defined builds
- fix license headers and remove some commented code

JochenKienzle committed:
Re-enable integration tests for cpp

- solve relative path issues (pg-config, adapters)
- copy pg-config files for basic example
- adapt paths and links (testset vs. testsuite)
- adapt pg-config for testsuites
- add build folders to dockerignore to avoid pre-defined builds
- fix license headers and remove some commented code

Signed-off-by: Jochen Kienzle <Jochen.Kienzle@de.bosch.com>

JochenKienzle commented:
@mikehaller @mariaivanova-git 
Ready for merge :) please check --> Would be cool to re-enable the cpp integration tests soon.

@mariaivanova-git If there are no findings--> Could you merge it then directly?

mariaivanova-git commented:
> @mikehaller @mariaivanova-git
> Ready for merge :) please check --> Would be cool to re-enable the cpp integration tests soon.
> 
> @mariaivanova-git If there are no findings--> Could you merge it then directly?

cpp test suites need to be added to startIntegratonTests.sh: at the moment they are referred by the config.json of the runner but the docker-compose files are not called
I'll add this and a few minor changes with a commit in this PR later

mariaivanova-git committed:
Add cpp docker-compose files in startIntegrationTests.sh
+ rename junit-report.xml to junit.xml
+ minor editorial changes like fixing tabs and whispaces
+ fix script licences to Bosch.IO header

Signed-off-by: Maria Ivanova <maria.ivanova@bosch.io>

JochenKienzle commented:
@mariaivanova-git Thanks for the fixes 👍  Looks good.
Do you need more time for the review?

mariaivanova-git commented on test/integration-tests/sdk-test-suites/cpp/function_provider.cpp:
```diff
@@ -38,12 +38,12 @@ class FunctionProvider : public FunctionTalent {
 static auto mqtt_config = json{
```
The configurations of the other test-suites are in separate config.json files located in test/integration-tests/config/tests/<language>. It would be good to have the same structure for cpp.

JochenKienzle replied:
Good point. There are other optimizations possible but this makes sense to provide flexible configurations.
I'll change

JochenKienzle replied:
Done :)

mariaivanova-git commented on test/integration-tests/sdk-test-suites/cpp/function_provider.cpp:
```diff
@@ -38,12 +38,12 @@ class FunctionProvider : public FunctionTalent {
     {"platform", true},
     {"module", {
-                   {"name", "./testset_sdk/cpp/build/iotea-sdk-cpp-lib/adapters/mqtt/libmqtt_protocol_adapter.so"}
+                   {"name", "./iotea-sdk-cpp-lib/adapters/mqtt/libmqtt_protocol_adapter.so"}
```
Same as above: The configurations of the other test-suites are in separate config.json files located in test/integration-tests/config/tests/<language>. It would be good to have the same structure for cpp.

JochenKienzle replied:
Good point. There are other optimizations possible but this makes sense to provide flexible configurations.
I'll change

JochenKienzle replied:
Done :)

JochenKienzle committed:
Rename TestSetSDK to TestSuiteSDK for cpp

Signed-off-by: Jochen Kienzle <Jochen.Kienzle@de.bosch.com>

JochenKienzle committed:
Improve cmake build for cpp integration tests

- Provide option for 'build examples'
- Improve dockerfiles to build only necessary binaries

Signed-off-by: Jochen Kienzle <Jochen.Kienzle@de.bosch.com>

JochenKienzle committed:
Remove hardcoded config in cpp integration tests

Signed-off-by: Jochen Kienzle <Jochen.Kienzle@de.bosch.com>

JochenKienzle commented:
@mariaivanova-git Can you check again? :)

mikehaller commented on test/integration-tests/sdk-test-suites/cpp/testsuite_sdk.cpp:
```diff
@@ -46,37 +47,26 @@ class TestSuiteSDK : public TestSuiteTalent {
+    client->RegisterTalent(talent);
 
-    client.Start();
+    std::signal(SIGINT, signal_handler);
```
SIGINT is for Ctrl-C, but e.g. edgecontainerd is sending SIGTERM when you ask it to stop the container.
Should it handle both signals?

JochenKienzle replied:
I have added SIGTERM as well

JochenKienzle committed:
Handle sigterm in sdk-test-suites cpp

Signed-off-by: Jochen Kienzle <Jochen.Kienzle@de.bosch.com>

mariaivanova-git committed:
Fix github action integration tests workflow to indicate test failure

Signed-off-by: Maria Ivanova <maria.ivanova@bosch.io>

JochenKienzle committed:
Fix config paths, linking and type for integration test cpp

Signed-off-by: Jochen Kienzle <Jochen.Kienzle@de.bosch.com>
